ntbs-service/Pages/Notifications/NotificationsBase.cshtml.cs

```diff
@@ -16,31 +24,126 @@ public abstract class NotificationModelBase : ValidationModel
         }
 
         public Notification Notification { get; set; }
+
         [BindProperty]
         public int? NotificationId { get; set; }
         
         // This can be thrown away once proper banner work completes
         public NotificationStatus NotificationStatus { get; set; }
 
+        [ViewData]
+        public Dictionary<string, NotifyError> NotifyErrorDictionary { get; set; }
+
         public async Task<IActionResult> OnPostSubmitAsync()
         {
+            // First Validate and Save current page details
             bool isValid = await ValidateAndSave(NotificationId);
-            if (!isValid) {
+            if (!isValid) 
+            {
                 return await OnGetAsync(NotificationId);
             }
-            
-            var notification = await service.GetNotificationAsync(NotificationId);
-            if (notification == null)
+
+            // Get Notifications with all owned properties to check for 
+            Notification = await service.GetNotificationWithAllInfoAsync(NotificationId);
+            if (Notification == null)
             {
                 return NotFound();
             }
+            
+            // IsRequired fields on Models requires ShouldValidateFull flag
+            SetShouldValidateFull();
+
+            if (!IsModelValid())
+            {
+                GetModelStateErrors();
+                return Partial("./NotificationErrorSummary", this);
+            } 
 
-            await service.SubmitNotification(notification);
+            await service.SubmitNotification(Notification);
             
             return RedirectToPage("../Overview", new {id = NotificationId});
         }
 
-        
+        private bool IsModelValid()
+        {
+            return TryValidateModel(Notification);
+        }
+
+        private void SetShouldValidateFull() 
+        {
+            Notification.ShouldValidateFull = true;

```

Comment:
I wonder whether it's possible with EF to iterate over all owned properties using reflection, and then just setting them all in a loop within notification instead of spelling this all out here?

Reply:
Good idea, I will have a look, I think there should be a way to iterate through owned properties